local-volume/README.md

```diff
@@ -165,30 +165,68 @@ $ kubectl create -f bootstrapper/deployment/kubernetes/example-storageclass.yaml
-sample configuration files.
+This option uses helm templating to generate the provisioner's DaemonSet and ConfigMap specs.
+The generated specs can be further customized as needed (usually not necessary), and then deployed using kubectl.
 
-1. Create an admin account with cluster admin priviledge:
+**helm template** uses 3 sources of information:
+1. Provisioner's chart template located at helm/provisioner/templates/provisioner.yaml
+2. Provisioner's default values.yaml which contains variables used for rendering a template.
+3. (Optional) User's customized values.yaml as a part of helm template command. User's provided
+   values will override default values of Provisioner's values.yaml.
+
+Default values.yaml is located in local-volume/helm/provisioner folder, user should not remove variables from this file but can
+change any values of these variables. 
+The only exception is the list of storage classes and respective mounting points. In general there should be at least one storage class
+listed with its corresponding "hostDir", "mountDir" parameter is optional in case and in case of absense, the value of "defaultMountDir"
+will be used.
+
+Follow these steps to bring up Provisioner and persistent local volumes.
+
+1. Install helm binary
+In order to be able to use **helm** to render templates, it has to be installed on a host where a user plans
+to generate templates.
+
+These two commands will download and copy helm binary into /usr/bin directory.
+
+``` console
+export HELM_URL=http://storage.googleapis.com/kubernetes-helm/helm-v2.7.2-linux-amd64.tar.gz
+curl "$HELM_URL" | sudo tar --strip-components 1 -C /usr/bin linux-amd64/helm -zxf -
+```
+Provisioner's spec generation process has been tested with helm version 2.7.2.
+
+2. Create an admin account with cluster admin priviledge:
 ``` console
-$ kubectl create -f bootstrapper/deployment/kubernetes/<version>/admin-account.yaml
+$ kubectl create -f local-volume/provisioner/deployment/kubernetes/admin_account.yaml  
 ```
 
-2. Create a ConfigMap with your local storage configuration details.
+3. Generate Provisioner spec and config map.
+In the simplest form, to generate Provisioner spec, you run:
+
 ``` console
-$ kubectl create -f bootstrapper/deployment/kubernetes/<version>/example-config.yaml
+helm template local-volume/helm/provisioner > provisioner.yaml
 ```
-**Note**: This configmap is required in order to run the bootstrapper.
 
+This command will generate a ready to use with **kubectl** yaml file. A user can further
+modify generated file, but then **kubectl** needs to be used to check validity of additional changes.
 
-3. Launch the bootstrapper, which in turn creates static provisioner daemonset:
+In case a user does not want to modify default values.yaml, a custom yaml file can be used in the command line:
+
+``` console
+helm template local-volume/helm/provisioner --values custom-values.yaml > provisioner.yaml
```

Comment:
Q1: Can we put the generated file inside a `deployment/kuberentes/<version>/` directory? I'm thinking it should probably go into `deployment/kuberentes/<version>/provisioner/`

Q2: With helm generated files, is it common to include` _generated` on filenames or some other indicator that this particular file has been generated by helm and that should be regenerated from the template and not modified directly?

Reply:
I used to have with version in the path but @msau42 requested to change it.

Reply:
I'm ok without version, but I think the generated file should land inside `deployment/kubernetes/<>/provisioner/` directory.